src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -492,28 +493,22 @@ void activity_classifier::init_train(
   }
   init_options(opts);
 
+  add_or_update_state({{"session_id", session_id_column_name},
+                       {"target", target_column_name},
+                       {"features", flex_list(feature_column_names.begin(),
+                                              feature_column_names.end())}});
+
   // Bind the data to a data iterator.
-  data_iterator::parameters data_params;
-  data_params.data = data;
-  data_params.target_column_name = target_column_name;
-  data_params.session_id_column_name = session_id_column_name;
-  data_params.feature_column_names = feature_column_names;
-  data_params.prediction_window = read_state<flex_int>("prediction_window");
-  data_params.predictions_in_chunk = NUM_PREDICTIONS_PER_CHUNK;
-  training_data_iterator_ = create_iterator(data_params);
+  training_data_iterator_ = create_iterator(data, {}, true);
```

Comment:
Hmm, I notice this call to `create_iterator` is the only one that provides an empty list of class labels, and it's the only one that passes `true` for `is_train`. Here's an idea: what if `create_iterator` just took the `data` and `is_train` arguments, and its implementation looks as `is_train` to determine whether to populate the `class_labels` parameter from the model state. (It can also then just set the param struct's `verbose` flag with the value of `is_train`.)